### PR TITLE
bugfix/notify-user-when-no-annotations-are-selected

### PR DESCRIPTION
**Pull request by CorreyL**
## Preamble

This PR implements a notification when a user has not selected any annotations to be exported.

## Screenshots

image

image

## Commit History

* Notify the user if no annotations will be exported

The previous logic would not notify the user if no annotations were
selected, thus exporting an empty JsonArray.

Fix the logic to determine if any annotations are selected for exporting
before the pop-up to select where to save the `.json` file

If no annotations are selected, display an error to the user and stop
the extension from executing any further logic

### CODE REVIEW

**Comment by CorreyL** [comment]
Worth noting that the follow chunk of code:

https://github.com/CorreyL/qupath-annotation-exchange/pull/4/files#diff-b262adcc80ae27cbd5e071cb869e9531R55-R62

Replicates logic from `ExportAnnotationServiceJSONPlugin.java`:

https://github.com/CorreyL/qupath-annotation-exchange/blob/dev/src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSONPlugin.java#L95-L104

File: src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java [hunk 1]
```diff
@@ -30,13 +37,38 @@ public ExportAnnotationServiceJSON(QuPathGUI qupath){
...
+    }
+
     public void run(){
         QuPathViewer viewer = qupath.getViewer();
         if (viewer == null || viewer.getServer() == null) {
             logger.error("No Slide Loaded.");
             return;
         }
 
+        PluginRunnerFX runner = new PluginRunnerFX(qupath,false);
```
Comment (areisle) [inline src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java:53]: is there an auto formatter for java we can run?

File: src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java [hunk 2]
```diff
@@ -30,13 +37,38 @@ public ExportAnnotationServiceJSON(QuPathGUI qupath){
...
+            .getHierarchy()
+            .getSelectionModel()
+            .getSelectedObjects();
+
+        Collection<Class<? extends PathObject>> supported = getSupportedParentObjectClasses();
+
+        Collection<? extends PathObject> objects = PathObjectTools.getSupportedObjects(selectedObjects, supported);
+
+        if (objects.size() == 0) {
+            DisplayHelpers.showErrorMessage(
```
Comment (areisle) [inline src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java:65]: this looks good!

**Review by areisle — APPROVED** [review]

File: src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java (hunk 1 above)
Comment (CorreyL) [reply src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSON.java:53]: I have `SonarLint` installed in my IDE:

https://www.sonarlint.org/intellij/

And I admittedly haven't looked very hard, but I haven't come across an equivalent to `.eslintrc` that I can commit.

I'll see if I have time to look a bit harder in a future PR so we can standardize the code style.

**Pull request merged**